### PR TITLE
[simple-object-detection] Allow some noise objects to obscure target object

### PR DESCRIPTION
**Pull request by caisq**
- Previously, all the background objects (line segments and circles) were drawn
  before the target objects and hence are in the background.
- In this PR, we allow 50% of the background objects to be drawn after the target
  objects. As a result, some of them will end up on to of the target objects and
  partially cover them.
- The model turns out to be able to handle it. There is an expected slight increase
  in the validation loss due to the partial covering, but the increase doesn't  lead
  to any serious degradation of the object classification or localization quality.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-examples/176)

### CODE REVIEW

**Comment by bileschi** [comment]
___
*[simple-object-detection/synthetic_images.js, line 158 at r1](https://reviewable.io/reviews/tensorflow/tfjs-examples/176#-LRHC6qQCGsTRBMGCJbB:-LRHC6qQCGsTRBMGCJbC:b71j32m) ([raw file](https://github.com/tensorflow/tfjs-examples/blob/c297113ec2123104f66e24a302aeff52f26334ad/simple-object-detection/synthetic_images.js#L158)):*
> ```
> (let i = 0; i < numCircles / 2; ++i) 
> ```
You could rid yourself of the off-by-one counting issue for odd numbers of circles by replacing this with:
```
(let i = numCircles/2; i < numCircles; ++i)
```
